daily-rotate-file.js

```diff
@@ -20,134 +18,167 @@ var loggerDefaults = {
+        return !/["<>|:*?\\/\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f]/g.test(
+            filename
+        );
     }
 
     function isValidDirName(dirname) {
         // eslint-disable-next-line no-control-regex
-        return !/["<>|\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f]/g.test(dirname);
+        return !/["<>|\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f]/g.test(
+            dirname
+        );
     }
 
     this.options = Object.assign({}, loggerDefaults, options);
 
     if (options.stream) {
-        throwIf('stream', 'filename', 'maxsize');
+        throwIf("stream", "filename", "maxsize");
         this.logStream = new PassThrough();
         this.logStream.pipe(options.stream);
     } else {
-        this.filename = options.filename ? path.basename(options.filename) : 'winston.log';
+        this.filename = options.filename
+            ? path.basename(options.filename)
+            : "winston.log";
         this.dirname = options.dirname || path.dirname(options.filename);
 
         if (!isValidFileName(this.filename) || !isValidDirName(this.dirname)) {
-            throw new Error('Your path or filename contain an invalid character.');
+            throw new Error("Your path or filename contain an invalid character.");
         }
 
-        var self = this;
-
-        this.logStream = require('file-stream-rotator').getStream({
+        this.logStream = require("file-stream-rotator").getStream({
             filename: path.join(this.dirname, this.filename),
-            frequency: options.frequency ? options.frequency : 'custom',
-            date_format: options.datePattern ? options.datePattern : 'YYYY-MM-DD',
+            frequency: options.frequency ? options.frequency : "custom",
+            date_format: options.datePattern ? options.datePattern : "YYYY-MM-DD",
             verbose: false,
             size: getMaxSize(options.maxSize),
             max_logs: options.maxFiles,
             end_stream: true,
-            audit_file: options.auditFile ? options.auditFile : path.join(self.dirname, '.' + hash(options) + '-audit.json'),
-            file_options: options.options ? options.options : {flags: 'a'},
+            audit_file: options.auditFile
+                ? options.auditFile
+                : path.join(this.dirname, "." + hash(options) + "-audit.json"),
+            file_options: options.options ? options.options : { flags: "a" },
             utc: options.utc ? options.utc : false,
-            extension: options.extension ? options.extension : '',
+            extension: options.extension ? options.extension : "",
```

Comment:
didn't now that you prefer this quote style :wink:

Reply:
In fact I mainly left the quoting as it was before ... just streamlined it to be consistent